### PR TITLE
feat(payment): PAYPAL-4607 Braintree Trustly payment method

### PR DESCRIPTION
**Pull request by bc-nick**
## What?

Braintree Trustly payment method

## Why?

To be able place an order via Braintree Trustly payment method

Depends on:

https://github.com/bigcommerce/checkout-sdk-js/pull/2653

https://github.com/bigcommerce/checkout-js/pull/2019

## Testing / Proof

https://github.com/user-attachments/assets/74c91832-b1e2-46ca-b3ae-2b462155bf74

@bigcommerce/team-checkout

### CODE REVIEW

**Review by andriiVitvitskyi1990 — APPROVED** [review]

File: packages/braintree-integration/src/validation-schemas/getBraintreeTrustlyValidationSchema.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    return object(
+        formFieldData.reduce((schema, { id, required }) => {
+            if (required) {
+                schema[id] = string()
+                    .nullable()
+                    .required(language.translate(`payment.errors.phone_number_required_error`));
+
+                if (id === 'phoneNumber') {
+                    schema[id] = schema[id].matches(
+                        /^\d{7,11}$/,
```
Comment (serhii-tkachenko) [inline packages/braintree-integration/src/validation-schemas/getBraintreeTrustlyValidationSchema.ts:21]: Why do 7 to 11 digits range for phone number?

**Review by serhii-tkachenko — APPROVED** [review]
Great job 👍

File: packages/braintree-integration/src/validation-schemas/getBraintreeTrustlyValidationSchema.ts (hunk 1 above)
Comment (bc-nick) [reply packages/braintree-integration/src/validation-schemas/getBraintreeTrustlyValidationSchema.ts:21]: This validation has been taken from Rate Pay validation scheme. But i need to rethink it

**Comment by bc-nick** [comment]
This PR was closed because of changed requirements

**Pull request closed without merging**